src/LocalInstaller.ts

```diff
@@ -55,22 +58,38 @@ export class LocalInstaller extends EventEmitter {
     }
 
     public async install(): Promise<InstallTarget[]> {
+        await this.createTmpDirectory(getTmpDir(this.uniqueDir));
+
         const packages = await this.resolvePackages();
         const installTargets = this.identifyInstallTargets(packages);
-        const packedSources = await this.packAll();
+
+        await this.packAll();
         await this.installAll(installTargets);
-        await this.removeAllPackedTarballs(packedSources, packages);
+        await this.removeTmpDirectory();
+
         return installTargets;
     }
 
+    public async createTmpDirectory(tmpDir: string) {
+        return new Promise((resolvePromise, rejectPromise) => {
+            fs.mkdir(tmpDir, (err) => {
```

Comment:
`fs.mkdir` already *returns a promise* (thanks to the magic of [mz](https://www.npmjs.com/package/mz)). So this method is not needed.